### PR TITLE
[web] define $flutterDriverResult variable early in driver test initalization

### LINKED ISSUE #102660: Linux web_long_running_tests_1_5 is 2.00% flaky
The post-submit test builder `Linux web_long_running_tests_1_5` had a flaky ratio 2.00% for the past (up to) 100 commits, which is above our 2.00% threshold.

One recent flaky example for a same commit: https://ci.chromium.org/ui/p/flutter/builders/prod/Linux%20web_long_running_tests_1_5/3845
Commit: https://github.com/flutter/flutter/commit/899548c48ae68eacadab3749ab9a0c46a1ffef93

Flaky builds:
https://ci.chromium.org/ui/p/flutter/builders/prod/Linux%20web_long_running_tests_1_5/3845
https://ci.chromium.org/ui/p/flutter/builders/prod/Linux%20web_long_running_tests_1_5/3808

Recent test runs:
https://flutter-dashboard.appspot.com/#/build?taskFilter=Linux%20web_long_running_tests_1_5

Please follow https://github.com/flutter/flutter/wiki/Reducing-Test-Flakiness#fixing-flaky-tests to fix the flakiness and enable the test back after validating the fix (internal dashboard to validate: go/flutter_test_flakiness).

### LINKED ISSUE #107545: Linux web_long_running_tests_4_5 is 2.00% flaky
The post-submit test builder `Linux web_long_running_tests_4_5` had a flaky ratio 2.00% for the past (up to) 100 commits, which is above our 2.00% threshold.

One recent flaky example for a same commit: https://ci.chromium.org/ui/p/flutter/builders/prod/Linux%20web_long_running_tests_4_5/5088
Commit: https://github.com/flutter/flutter/commit/1704d4f5f9c2f53decd1d526a8a41dfa06d1e484

Flaky builds:
https://ci.chromium.org/ui/p/flutter/builders/prod/Linux%20web_long_running_tests_4_5/5088
https://ci.chromium.org/ui/p/flutter/builders/prod/Linux%20web_long_running_tests_4_5/5083

Recent test runs:
https://flutter-dashboard.appspot.com/#/build?taskFilter=Linux%20web_long_running_tests_4_5

Please follow https://github.com/flutter/flutter/wiki/Reducing-Test-Flakiness#fixing-flaky-tests to fix the flakiness and enable the test back after validating the fix (internal dashboard to validate: go/flutter_test_flakiness).

### PR DESCRIPTION
**Pull request by yjbanov**
Define `$flutterDriverResult` during driver VM service extension initialization, so that by the time `web_driver.dart` is checking for its value it is guaranteed to be available. Otherwise, there's a race between the time the variable is checked, and the time the first driver command initializes it.

Fixes https://github.com/flutter/flutter/issues/102660
Fixes https://github.com/flutter/flutter/issues/107545

### CODE REVIEW

**Comment by ditman** [comment]
Linked https://github.com/flutter/flutter/issues/107545 as well, that looks like a dup of the original issue this was fixing (in a different shard)

File: packages/flutter_driver/lib/src/driver/web_driver.dart [hunk 1]
```diff
@@ -294,6 +294,29 @@ class FlutterWebConnection {
...
   /// Sends command via WebDriver to Flutter web application.
   Future<dynamic> sendCommand(String script, Duration? duration) async {
+    // This code should not be reachable before the VM service extension is
+    // initialized. The VM service extension is expected to initialize both
+    // `$flutterDriverResult` and `$flutterDriver` variables before attempting
+    // to send commands. This part checks that `$flutterDriverResult` is present.
+    // `$flutterDriver` is not checked because it is covered by the `script`
+    // that's executed next.
+    try {
+      await _driver.execute(r'return $flutterDriverResult', <String>[]);
```
Comment (ditman) [inline packages/flutter_driver/lib/src/driver/web_driver.dart:304]: Have you considered wrapping this `await` in a `waitFor` call, similar to here:

https://github.com/flutter/flutter/blob/549f70c9146c5286f4aeca163724b4fbb402eec0/packages/flutter_driver/lib/src/driver/web_driver.dart#L304-L308

(I'd suggest a much shorter duration than 30 days, though)

Just to have some sort of watchdog that the await is not going to hang "forever" if it never ends up throwing?

File: packages/flutter_driver/lib/src/extension/_extension_web.dart [hunk 2]
```diff
@@ -18,6 +18,11 @@ import 'dart:js_util' as js_util;
 ///
 ///  * [_extension_io.dart], which has the dart:io implementation
 void registerWebServiceExtension(Future<Map<String, dynamic>> Function(Map<String, String>) call) {
+  // Define the result variable because packages/flutter_driver/lib/src/driver/web_driver.dart
+  // checks for this value to become non-null when waiting for the result. If this value is
+  // undefined at the time of the check, WebDriver throws an exception.
+  context[r'$flutterDriverResult'] = null;
```
Comment (ditman) [inline packages/flutter_driver/lib/src/extension/_extension_web.dart:24]: Why not use the same pattern as in line 26 to set this value?

```
js_util.setProperty(html.window, r'$flutterDriverResult', null);
```

File: packages/integration_test/lib/_extension_web.dart [hunk 3]
```diff
@@ -17,6 +17,11 @@ import 'dart:js_util' as js_util;
 ///
 ///  * `_extension_io.dart`, which has the dart:io implementation
 void registerWebServiceExtension(Future<Map<String, dynamic>> Function(Map<String, String>) callback) {
+  // Define the result variable because packages/flutter_driver/lib/src/driver/web_driver.dart
+  // checks for this value to become non-null when waiting for the result. If this value is
+  // undefined at the time of the check, WebDriver throws an exception.
+  context[r'$flutterDriverResult'] = null;
```
Comment (ditman) [inline packages/integration_test/lib/_extension_web.dart:23]: (Same comment as above)

**Review by ditman — APPROVED** [review]
Only little nitpicks with the js-interop style, and maybe adding a watchdog to the self-check bits of the `sendCommand` method.

File: packages/flutter_driver/lib/src/driver/web_driver.dart (hunk 1 above)
Comment (yjbanov) [reply packages/flutter_driver/lib/src/driver/web_driver.dart:304]: This is effectively removing a timeout, which is something we've been doing across the entire codebase. We already have global test timeouts that should be sufficient. If it hangs, we'll know there a bug and fix it.

File: packages/flutter_driver/lib/src/extension/_extension_web.dart (hunk 2 above)
Comment (yjbanov) [reply packages/flutter_driver/lib/src/extension/_extension_web.dart:24]: This is using the same pattern as in line 31 :) I'm not sure what the difference is between window and context. I think in a window context in the browser (as opposed to a web worker, service worker, or node.js server) there is no difference.

Comment (ditman) [reply packages/flutter_driver/lib/src/extension/_extension_web.dart:24]: Ah, I think this is mixing js-interop styles, that's confusing!

File: packages/flutter_driver/lib/src/driver/web_driver.dart (hunk 1 above)
Comment (ditman) [reply packages/flutter_driver/lib/src/driver/web_driver.dart:304]: The waitFor I suggested is not the one from flutter_driver, but from `'package:webdriver/support/async.dart'`

**Pull request merged**